packages/internal/bridge/bridge-sample-app/src/deposit.ts

```diff
@@ -27,7 +38,141 @@ async function deposit() {
+
+  let rootBridgeChildAddress = await rootBridge.rootTokenToChildToken(params.sepoliaToken);
+  let childBridgeChildAddress = await childBridge.rootTokenToChildToken(params.sepoliaToken);
+
+  if (rootBridgeChildAddress === ethers.constants.AddressZero) {
+    console.log('token not mapped, please map token before depositing');
+    return;
+  }
+
+  if (childBridgeChildAddress === ethers.constants.AddressZero) {
+    console.log('token mapping incomplete, please wait for token to map to childBridge before depositing');
+    return;
+  }
+
+  if (rootBridgeChildAddress !== childBridgeChildAddress) {
+    console.log(`token mappings mismatch on rootBridge (${rootBridgeChildAddress}) & childBridge (${childBridgeChildAddress}).`, );
+    return;
+  }
+
+  const approvalReq: ApproveBridgeRequest = {
+    senderAddress: params.sender,
+    token: params.sepoliaToken,
+    amount: params.amount,
+    sourceChainId: ETH_SEPOLIA_CHAIN_ID,
+    destinationChainId: ZKEVM_TESTNET_CHAIN_ID,
+  }
+
+  console.log('approvalReq', approvalReq);
+
+  let approvalRes: ApproveBridgeResponse;
+  try {
+    approvalRes = await tokenBridge.getUnsignedApproveBridgeTx(approvalReq);
+    console.log('approvalRes', approvalRes);
+  } catch(err) {
+    console.log('approvalErr', err);
+    return
+  }
+
+  if (approvalRes!.unsignedTx) {
+    const approvalNonce = await params.rootWallet.getTransactionCount();
+    const approvalGasPrice = await params.rootProvider.getGasPrice();
+
+    console.log('approvalNonce', approvalNonce);
+    console.log('approvalGasPrice', approvalGasPrice);
+
+    approvalRes!.unsignedTx.gasLimit = BridgeMethodsGasLimit.DEPOSIT_SOURCE;
+    approvalRes!.unsignedTx.nonce = approvalNonce;
+    approvalRes!.unsignedTx.gasPrice = approvalGasPrice.mul(2);
+
+    console.log('approvalRes.unsignedTx');
+    console.log(approvalRes!.unsignedTx);
+
+    console.log('signing approval');
+    const approvalTxSig = await params.rootWallet.signTransaction(approvalRes!.unsignedTx);
+    console.log('approvalTxSig', approvalTxSig);
+
+    const sendApprovalRes = await params.rootWallet.provider.sendTransaction(approvalTxSig);
+    console.log('sendApprovalRes', sendApprovalRes);
+
+    await waitForReceipt(sendApprovalRes.hash, params.rootProvider);
+  } else {
+    console.log('no approval required');
+  }
+
+  const depositReq: BridgeTxRequest = {
+    senderAddress: params.sender,
+    recipientAddress: params.recipient,
+    token: params.sepoliaToken,
+    amount: params.amount,
+    sourceChainId: ETH_SEPOLIA_CHAIN_ID,
+    destinationChainId: ZKEVM_TESTNET_CHAIN_ID,
+    gasMultiplier: 1.1,
+  }
+
+  console.log('depositReq', depositReq)
+  let depositRes: BridgeTxResponse;
+  try {
+    depositRes = await tokenBridge.getUnsignedBridgeTx(depositReq);
+    console.log('depositRes', depositRes);
+  } catch(err) {
+    console.log('depositErr', err);
```

Comment:
should we use `console.error()` here, and other similar places (i.e. inside catch clauses)?